src/databroker_elasticsearch/converters.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+"""\
+Functions for converting values that are exported to Elasticsearch.
+"""
+
+# handle registration of converter functions ---------------------------------
+
+def register_converter(f, name=None):
+    "Decorator to mark up some function as a converter."
+    nm = f.__name__ if name is None else name
+    _converters[nm] = f
+    return f
+
+_converters = {}
+
+
+def getconverter(name):
+    "Return converter function of the specified name."
+    return _converters[name]
+
+# define and register converter functions ------------------------------------
+
+register_converter(int)
+register_converter(float)
+register_converter(str)
+
+
+@register_converter
+def noconversion(x):
+    "Return the argument as is."
+    return x
+
+
+@register_converter
+def toisoformat(epoch):
+    """Convert epoch seconds to elasticsearch friendly ISO time.
+
+    When `epoch` is a float return ISO date with millisecond
+    precision.  Otherwise return date rounded to seconds.
+
+    Parameters
+    ----------
+    epoch : float
+        The time in seconds since POSIX epoch in 1970.
+
+    Returns
+    -------
+    isodate : str
+        The ISO formatted date and time with second or millisecond precision.
+    """
+    from datetime import datetime
+    epochms = round(epoch, 3)
+    dt = datetime.fromtimestamp(epochms)
+    tiso = dt.isoformat()
+    rv = tiso[:-3] if dt.microsecond else tiso
+    assert len(rv) in (19, 23)
+    return rv
+
+
+@register_converter
+def normalize_counts(d):
+    """Normalize numeric values in a dictionary to a total of 1.
+
+    For obtaining normalized stoichiometry in a chemical formula.
+
+    Parameters
+    ----------
+    d : dict or collections.OrderedDict
+        The dictionary to be normalized.
+
+    Returns
+    -------
+        A copy of `d` with values summing to a total of 1.
+        Return ``None`` when `d` is of invalid type.
+    """
+    if not isinstance(d, dict):
```

Comment:
Should this check against ``MutableMap``?

Reply:
Is there any `MutableMap` type that is not a `dict` in use in bluesky, mongodb or elsewhere in NSLS-II sources?
If not let's cross that bridge when we get to it.

Reply:
Usually it is better form to compare against the ABC just in case. We got bit by this with conda-forge when `ruamle.yaml` changed their types from dict to `MutableMapping`

Reply:
OK will do.